docs/usage/TYPE_SYNC.md

```diff
@@ -0,0 +1,117 @@
+streams would look as follows:
+```java
+final Logger logger = LoggerFactory.getLogger(MySync.class);
+final TypeSyncOptions typeSyncOptions = TypeSyncOptionsBuilder.of(sphereClient)
+                                                              .errorCallBack(logger::error)
+                                                              .warningCallBack(logger::warn)
+                                                              .build();
+```
+
+#### Running the sync
+After all the aforementioned points in the previous section have been fulfilled, to run the sync:
+````java
+// instantiating a type sync
+final TypeSync typeSync = new TypeSync(typeSyncOptions);
+
+// execute the sync on your list of types
+CompletionStage<TypeSyncStatistics> syncStatisticsStage = typeSync.sync(typeDrafts);
+````
+The result of the completing the `syncStatisticsStage` in the previous code snippet contains a `TypeSyncStatistics`
+which contains all the stats of the sync process; which includes a report message, the total number of updated, created,
+failed, processed types and the processing time of the last sync batch in different time units and in a
+human-readable format.
+
+````java
+final TypeSyncStatistics stats = syncStatisticsStage.toCompletebleFuture().join();
+stats.getReportMessage();
+/*"Summary: 2000 types were processed in total (1000 created, 995 updated, 5 failed to sync)."*/
+````
+
+__Note__ The statistics object contains the processing time of the last batch only. This is due to two reasons:
+ 1. The sync processing time should not take into account the time between supplying batches to the sync.
+ 2. It is not known by the sync which batch is going to be the last one supplied.
+
+More examples of how to use the sync can be found [here](/src/integration-test/java/com/commercetools/sync/integration/types/TypeSyncIT.java).
+
+### Build all update actions
+
+A utility method provided by the library to compare a `Type` with a new `TypeDraft` and results in a list of type update actions.
+```java
+List<UpdateAction<Type>> updateActions = TypeSyncUtils.buildActions(type, typeDraft, typeSyncOptions);
+```
+
+### Build particular update action(s)
+
+Utility methods provided by the library to compare the specific fields of a `Type` and a new `TypeDraft`, and in turn builds
+ the update action. One example is the `buildChangeNameUpdateAction` which compares names:
+````java
+Optional<UpdateAction<Type>> updateAction = TypeUpdateActionUtils.buildChangeNameAction(oldType, typeDraft);
+````
+More examples of those utils for different types can be found [here](/src/test/java/com/commercetools/sync/types/utils/TypeUpdateActionUtilsTest.java).
+and field definitions can be found [here](/src/test/java/com/commercetools/sync/types/utils/FieldDefinitionUpdateActionUtilsTest.java).
+
+## Caveats
+
+1. Types are either created or updated. Currently, the tool does not support type deletion.
```

Comment:
Maybe we should already remove this line from here. Since we decided not to have it as a caveat ;) But if you don't want to remove it in this PR. We could remove it with the rest (other sync readmes) in another PR..

Reply:
I fixed it here and added the SetType as a caveat also.  https://github.com/commercetools/commercetools-sync-java/pull/333/commits/fbb2d52fe2921d358c5775d453a331de27851971